src/__tests__/ceramic_integration.test.ts

```diff
@@ -0,0 +1,330 @@
+    const doc2 = await TileDocument.create(ceramic2, {foo: 1}, null, { anchor: true })
+
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+    expect(doc2.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+
+    // Test that anchoring on CAS1 doesn't anchor requests made against CAS2
+    await anchorUpdate(doc1, cas1)
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+    expect(doc2.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+
+    // Now test that anchoring on CAS2 doesn't anchor requests made against CAS1
+    await doc1.update({foo:2}, null, { anchor: true })
+    await anchorUpdate(doc2, cas2)
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+    expect(doc2.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+  }, 60 * 1000 * 3);
+
+  test('Multiple anchors for same stream', async () => {
+    const doc1 = await TileDocument.create(ceramic1, {foo: 1}, null, { anchor: true })
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+    await anchorUpdate(doc1, cas1)
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+
+    // Now that genesis commit has been anchored do an update and make sure anchoring works again
+    await doc1.update({foo:2}, null, { anchor: true })
+    await anchorUpdate(doc1, cas1)
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+    expect(doc1.content).toEqual({foo:2})
+  }, 60 * 1000 * 3);
+
+  test('Consensus for anchors', async () => {
+    const initialContent = { foo: 0 }
+    const updatedContent = { foo: 1 }
+
+    const doc1 = await TileDocument.create(ceramic1, initialContent, null, { anchor: true })
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.PENDING)
+
+    const doc2 = await TileDocument.load(ceramic2, doc1.id)
+    await doc2.update(updatedContent, null, { anchor: true })
+
+    await doc1.sync({sync: SyncOptions.SYNC_ALWAYS})
+    expect(doc2.content).toEqual(updatedContent)
+    expect(doc1.content).toEqual(updatedContent)
+
+    const docOnCas1Ceramic = await TileDocument.load(casCeramic1, doc1.id)
+    const docOnCas2Ceramic = await TileDocument.load(casCeramic2, doc1.id)
+    expect(docOnCas1Ceramic.content).toEqual(updatedContent)
+    expect(docOnCas2Ceramic.content).toEqual(updatedContent)
+
+    // At this point we have verified that *all* involved ceramic nodes (both user-facing ceramic
+    // nodes as well as both ceramic nodes used by the 2 anchor services) all know about the newest
+    // version of the stream.
+    await anchorUpdate(doc1, cas1)
+    expect(doc1.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+
+    // This fails! Even though all Ceramic nodes know about the new update, 'cas1' still anchors
+    // the genesis contents that it was told to anchor.
+    // TODO(#253): Uncomment this once CAS consensus rules are improved
+    //expect(doc1.content).toEqual(updatedContent)
```

Comment:
this is the actual thing I set out to reproduce with this test.  This is the piece that gets fixed with the CAS Consensus Improvements project.  The PR that actually fixes the behavior will uncomment this line